src/module.c

```diff
@@ -5783,7 +5783,17 @@ RedisModuleCallReply *RM_Call(RedisModuleCtx *ctx, const char *cmdname, const ch
 
     if (flags & REDISMODULE_ARGV_RESPECT_DENY_OOM) {
         if (cmd->flags & CMD_DENYOOM) {
-            if (server.pre_command_oom_state) {
+            int oom_state;
+            if (ctx->flags & REDISMODULE_CTX_THREAD_SAFE) {
+                /* On background thread we can not count on server.pre_command_oom_state.
+                 * Because it is only set on the main thread, in such case we will check
+                 * the actual memory usage. */
+                float level;
+                oom_state = (getMaxmemoryState(NULL,NULL,NULL,&level) == C_ERR);
```

Comment:
This function is faster when level is NULL, and we don't need it.
```suggestion
                oom_state = (getMaxmemoryState(NULL,NULL,NULL,NULL) == C_ERR);
```